src/main/java/minijava/ir/optimize/JmpBlockRemover.java

```diff
@@ -20,14 +19,14 @@ public boolean optimize(Graph graph) {
     this.graph = graph;
     this.hasChanged = false;
     BackEdges.enable(graph);
-    graph.walkTopological(this);
+    graph.walkPostorder(this);
     BackEdges.disable(graph);
     return hasChanged;
   }
 
   @Override
   public void visit(Block block) {
-    if (!hasChanged && isJmpBlock(block) && hasOnePredecessor(block)) {
+    if (isJmpBlock(block) && hasOnePredecessor(block)) {
```

Comment:
Also I don't think that we need the topological order at all, since we don't actually change any control flow paths. Well, it's mostly by stomach telling me that it's safe.

I changed it in the first place because `graph.walk*` variants would not visit all nodes, which is really strange and might have to do with the interplay of some global state in firm, who knows...